faker.go

```diff
@@ -318,86 +215,40 @@ func ResetUnique() {
 	uniqueValues = map[string][]interface{}{}
 }
 
-// SetGenerateUniqueValues allows to set the single fake data generator functions to generate unique data.
-func SetGenerateUniqueValues(unique bool) {
-	generateUniqueValues = unique
+func initMapperTagWithOption(opts ...options.OptionFunc) {
+	mapperTag[EmailTag] = GetNetworker(opts...).Email
+	mapperTag[MacAddressTag] = GetNetworker(opts...).MacAddress
+	mapperTag[DomainNameTag] = GetNetworker(opts...).DomainName
+	mapperTag[URLTag] = GetNetworker(opts...).URL
+	mapperTag[UserNameTag] = GetNetworker(opts...).UserName
+	mapperTag[IPV4Tag] = GetNetworker(opts...).IPv4
+	mapperTag[IPV6Tag] = GetNetworker(opts...).IPv6
+	mapperTag[PASSWORD] = GetNetworker(opts...).Password
+	mapperTag[JWT] = GetNetworker(opts...).Jwt
 }
 
-// SetIgnoreInterface allows to set a flag to ignore found interface{}s.
-func SetIgnoreInterface(ignore bool) {
-	ignoreInterface = ignore
-}
-
-// SetNilIfLenIsZero allows to set nil for the slice and maps, if size is 0.
-func SetNilIfLenIsZero(setNil bool) {
-	shouldSetNil = setNil
-}
-
-// SetRandomStringLength sets a length for random string generation
-func SetRandomStringLength(size int) error {
-	if size < 0 {
-		return fmt.Errorf(ErrSmallerThanZero, size)
-	}
-	randomStringLen = size
-	return nil
-}
-
-// SetStringLang sets language of random string generation (LangENG, LangCHI, LangRUS, LangJPN, LangKOR, EmotEMJ)
-func SetStringLang(l langRuneBoundary) {
-	lang = l
-}
-
-// SetRandomMapAndSliceSize sets the size for maps and slices for random generation.
-// deprecates, currently left for old version usage
-func SetRandomMapAndSliceSize(size int) error {
-	return SetRandomMapAndSliceMaxSize(size)
-}
-
-// SetRandomMapAndSliceMaxSize sets the max size for maps and slices for random generation.
-func SetRandomMapAndSliceMaxSize(size int) error {
```

Comment:
I prefer to keep those set function, it provides global setting and back forward compatibility

Reply:
I don't this is possible, since we remove the global variables, and using options

Reply:
And since we're introducing major version, breaking changes is acceptable. 

And no need to be 100% backward compatibility